### ISSUE TITLE
Allow chaining of plugins

### ISSUE THREAD

**Issue by aaronfrost** [scullyio/scully#97]
# 🧩 Feature request
If I have two plugins that I want to use on the same files, I should be able to chain them. 

### Description
If I have a route in my `scully.config.js` that is configured as such:
```
'blog/:slugId' : {
  type: 'contentFolder`,
  slugId: {
    folder: './blog'
  }
}
```
And I want each of the markdown files in that directory to be processed through the `contentFolder` render plugin AND ALSO a second render plugin that I created, that should be an option. 

Imagine that I had a render plugin that transformed the rendered HTML and replaced `:)` with `😊`, and I want that plugin to be ran AFTER the `contentFolder` plugin, I should be able to do this. 

### Describe the solution you'd like
```
'blog/:slugId' : {
  type: ['contentFolder', 'smiley'],
  slugId: {
    folder: './blog'
  }
}
```
### Describe alternatives you've considered
I realize that I can import the `contentFolder` plugin in the `smiley` plugin setup, and then run it and chain it manually. But it is more useful if we allow people to dynamically build these configurations inside of the `scully.config.js`. 

I could write the `smiley` render plugin like this:
```ecmascript 6
function smileyRenderPlugin(html, route) {
    return contentRenderPlugin(html, route).then(html => {
        return html.replace(/\:\)/g, '😊');
    })
}
registerPlugin('render', 'smiley', smileyRenderPlugin);
```
If I took this and then used `type: 'smiley'` in the route config, I run into several additional problems. 

First, while this would allow me, the plugin author, to compose a new plugin from other plugins, it doesn't allow the plugin consumer to compose them together. But I want to allow the plugin consumers to be able to compose these as well. Much in the same way that a webpack consumer can compose the uses of webpack plugins. 

Second, if I compose together in this way, then I have to make an accompanying `route plugin` as well. The `contentFolder` plugin has both a registered `router plugin` as well as a `render plugin`. If I create a `smiley render plugin`, I also have to create a `smiley router plugin`. I can totally do this, but it isn't as user friendly as I would like. 

We need this system to be the most user-friendly SSG in the history of them. We need to allow them to be composable.

**Comment by SanderElias** [comment]
Chaining plugins would only make sense for render-plugins, right?
I would like to propose something like this:
```js
'blog/:slugId' : {
  type: 'contentFolder', 
  postRender: ['smiley'],
  slugId: {
    folder: './blog'
  }
}
```
which would also open the path of en eventual `preRender` option.
Also, this would make it more easy to use things lie HTML optimization tools and so on.

**Comment by SanderElias** [comment]
closed by pr #160
